src/components/IndexTable/IndexTable.tsx

```diff
@@ -118,21 +125,57 @@ function IndexTableBase({
     );
   }, [handleSelectionChange, selectedItemsCount]);
 
-  const resizeTableHeadings = useCallback(() => {
-    if (!tableElement.current || !scrollableContainerElement.current) {
-      return;
-    }
+  const calculateFirstHeaderOffset = useCallback(() => {
+    return condensed
+      ? tableHeadingRects.current[0].offsetWidth
+      : tableHeadingRects.current[0].offsetWidth +
+          tableHeadingRects.current[1].offsetWidth;
+  }, [condensed]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const resizeTableHeadings = useCallback(
+    throttle(() => {
```

Comment:
one edge case to consider: this throttle will get reinitialized every time the callback gets reinitialized (which might not be an issue, since it's only dep is another callback). if this becomes an issue, one solution would be to store the throttle in a ref, so it doesn't change; the downside is that anything it references would also need to be a ref, but that might already be the case.